titan-server/src/accounts/excuses.rs

```diff
@@ -1,91 +1,142 @@
 }
 
 /// Queries the last inserted event excuse.
-pub fn find_most_recent(
+pub fn find_last_inserted(
     titan_db: &MysqlConnection
-) -> Result<models::UserEventExcuseWithType, diesel::result::Error> {
+) -> Result<models::UserEventExcuse, diesel::result::Error> {
     let excuse = schema::user_event_excuses::table
-        .inner_join(schema::event_types::table)
-        .select((
-            schema::user_event_excuses::all_columns,
-            schema::event_types::all_columns
-        ))
         .order_by(schema::user_event_excuses::id.desc())
         .first(titan_db)?;
 
-    Ok(map_excuse_with_type(excuse))
+    Ok(excuse)
+}
+
+pub fn find_all_unacknowledged(
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> QueryResult<Vec<models::UserEventExcuseWithAssoc>> {
+    let excuses = schema::user_event_excuses::table
+        .filter(schema::user_event_excuses::ack_date.is_null())
+        .order_by(schema::user_event_excuses::event_date.desc())
+        .get_results(titan_db)?;
+
+    map_excuses_assoc(excuses, titan_db, wcf_db, app_config)
 }
 
 /// Creates a new event excuse.
 pub fn create_event_excuse(
     new_event_excuse: &models::NewUserEventExcuse,
-    titan_db: &MysqlConnection
-) -> Result<models::UserEventExcuseWithType, diesel::result::Error> {
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> Result<models::UserEventExcuseWithAssoc, diesel::result::Error> {
     diesel::insert_into(schema::user_event_excuses::table)
         .values(new_event_excuse)
         .execute(titan_db)?;
 
-    find_most_recent(titan_db)
+    let excuse = find_last_inserted(titan_db)?;
+    map_excuse_assoc(excuse, titan_db, wcf_db, app_config)
 }
 
-fn map_excuses_with_type(
-    excuses: Vec<(models::UserEventExcuse, models::EventType)>
-) -> Vec<models::UserEventExcuseWithType> {
-    let mut excuses_with_type: Vec<models::UserEventExcuseWithType> = vec![];
-    for excuse in excuses {
-        excuses_with_type.push(models::UserEventExcuseWithType {
-            id: excuse.0.id,
-            user_id: excuse.0.id,
-            event_type: excuse.1,
-            event_date: excuse.0.event_date,
-            comments: excuse.0.comments,
-            ack_user_id: excuse.0.ack_user_id,
-            ack_date: excuse.0.ack_date,
-            ack_comments: excuse.0.ack_comments,
-            date_created: excuse.0.date_created,
-            date_modified: excuse.0.date_modified
-        });
-    }
+pub fn ack_event_excuse(
+    excuse_id: i32,
+    ack_user: &models::User,
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> QueryResult<models::UserEventExcuseWithAssoc> {
+    let mut excuse = find_by_id(excuse_id, titan_db)?;
+
+    excuse.ack_user_id = Some(ack_user.id);
+    excuse.ack_date = Some(chrono::Utc::now().naive_utc());
+
+    diesel::update(&excuse)
```

Comment:
The Rust compiler issues a warning here that there's an unused result being emitted by the call to `diesel::update`. I think this is just an instance of missing a `?` being applied at the end of this statement.